### PR TITLE
Make Element Call screensharing work on desktop

### PR DESCRIPTION
**Pull request by SimonBrandner**
Requires https://github.com/vector-im/element-call/pull/645
Fixes https://github.com/vector-im/element-web/issues/23382

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Make Element Call screensharing work on desktop ([\#9476](https://github.com/matrix-org/matrix-react-sdk/pull/9476)). Fixes vector-im/element-web#23382.

### CODE REVIEW

**Review by dbkr — APPROVED** [review]

File: src/models/Call.ts [hunk 1]
```diff
@@ -818,6 +816,7 @@ export class ElementCall extends Call {
         this.messaging!.on(`action:${ElementWidgetActions.HangupCall}`, this.onHangup);
         this.messaging!.on(`action:${ElementWidgetActions.TileLayout}`, this.onTileLayout);
         this.messaging!.on(`action:${ElementWidgetActions.SpotlightLayout}`, this.onSpotlightLayout);
+        this.messaging!.on(`action:${ElementWidgetActions.Screenshare}`, this.onScreenshare);
```
Comment (robintown) [inline src/models/Call.ts:819]: This needs a matching `off`

**Review by robintown — CHANGES_REQUESTED** [review]

**Comment by DoM1niC** [comment]
Doesn't fix the Element Desktop 
`Enabling screensharing error DOMException: Permission denied`

**Comment by SimonBrandner** [comment]
> Doesn't fix the Element Desktop `Enabling screensharing error DOMException: Permission denied`

It does work for me, what version of EC are you using?

File: src/models/Call.ts [hunk 2]
```diff
@@ -830,8 +830,9 @@ export class ElementCall extends Call {
     public setDisconnected() {
         this.client.off(ClientEvent.ToDeviceEvent, this.onToDeviceEvent);
         this.messaging!.off(`action:${ElementWidgetActions.HangupCall}`, this.onHangup);
-        this.messaging!.on(`action:${ElementWidgetActions.TileLayout}`, this.onTileLayout);
-        this.messaging!.on(`action:${ElementWidgetActions.SpotlightLayout}`, this.onSpotlightLayout);
```
Comment (robintown) [inline src/models/Call.ts:835]: whoops

**Review by robintown — APPROVED** [review]

**Pull request merged**